# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/x11/xfree86/devfs-vc.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- xc/programs/Xserver/hw/tinyx/linux/linux.c	2004-06-27 13:01:19.000000000 +0200
+++ xc-p/programs/Xserver/hw/tinyx/linux/linux.c	2004-11-11 22:20:32.022231840 +0100
@@ -125,10 +125,10 @@
 	vtno = kdVirtualTerminal;
     else
     {
-	if ((fd = open("/dev/tty0",O_WRONLY,0)) < 0) 
+	if ((fd = open("/dev/vc/0",O_WRONLY,0)) < 0) 
 	{
 	    FatalError(
-		       "LinuxInit: Cannot open /dev/tty0 (%s)\n",
+		       "LinuxInit: Cannot open /dev/vc/0 (%s)\n",
 		       strerror(errno));
 	}
 	if ((ioctl(fd, VT_OPENQRY, &vtno) < 0) ||
@@ -139,7 +139,7 @@
 	close(fd);
     }
 
-    sprintf(vtname,"/dev/tty%d",vtno); /* /dev/tty1-64 */
+    sprintf(vtname,"/dev/vc/%d",vtno); /* /dev/vc/1-64 */
 
     if ((LinuxConsoleFd = open(vtname, O_RDWR|O_NDELAY, 0)) < 0)
     {
@@ -156,7 +156,7 @@
      *
      * Why is this needed?
      */
-    LinuxCheckChown ("/dev/tty0");
+    LinuxCheckChown ("/dev/vc/0");
     /*
      * Linux doesn't switch to an active vt after the last close of a vt,
      * so we do this ourselves by remembering which is active now.
@@ -483,7 +483,7 @@
 	activeVT = -1;
     }
     close(LinuxConsoleFd);                /* make the vt-manager happy */
-    fd = open ("/dev/tty0", O_RDWR|O_NDELAY, 0);
+    fd = open ("/dev/vc/0", O_RDWR|O_NDELAY, 0);
     if (fd >= 0)
     {
 	ioctl (fd, VT_GETSTATE, &vts);
